blockchain/event_filter.go

```diff
@@ -26,13 +24,12 @@ type EventFilterer interface {
 }
 
 type EventFilter struct {
-	txn             db.Transaction
-	fromBlock       uint64
-	toBlock         uint64
-	contractAddress *felt.Felt
-	keys            [][]felt.Felt
-	maxScanned      uint // maximum number of scanned blocks in single call.
-	pendingBlockFn  func() *core.Block
+	txn            db.Transaction
+	fromBlock      uint64
+	toBlock        uint64
+	matcher        *EventMatcher
```

Comment:
I would also have `EventMatcher` here instead of `*EventMatcher` since it is not meant to be shared or be optional. Wdyt?

Reply:
Agree, the fields are just a pointer and a slice anyway. Let me make a quick change.